ledger/src/bank_forks.rs

```diff
@@ -218,6 +239,15 @@ impl BankForks {
         );
     }
 
+    fn check_interval(root: Slot, parent_slot: Slot, snapshot_interval_slots: usize) -> bool {
+        let interval = snapshot_interval_slots as u64;
+        assert!(parent_slot < root || root == 0);
+        assert!(interval > 0);
+        (root - parent_slot > interval)
+            || (root % interval == 0)
```

Comment:
do you need to check `(root % interval == 0)` if you changed the first check to `>=`? The only difference without this check is you would not generate banks on multiples of `interval` if something within the last `interval` was snapshotted. For instance, if the interval was `10` and the roots were:

`9, 19, 20, 21`, with this current code, you would snapshot `19, 20`, but with the proposed change you would snapshot `19`

Reply:
Actually even with your suggested change, this sequence still snapshots 19 and 20.

19%10 = 9, 20%10=0
9 >= 0 so it crosses the threshold of `(root % interval <= parent_slot % interval)`

Reply:
I'm liking `block_height % interval == 0` more and more..

Reply:
Granted to get this pattern you would have to skip the whole interval worth of slots.

Reply:
@sakridge yeah it seems if you want to guarantee that this snapshot and last snapshot are at least interval apart, you can't just look at the parent, you need to track a `last_snapshot` and also process the banks in order (so everybody comes to the same results about what the `last_snapshot` is)